pcdsutils/requirements.py

```diff
@@ -0,0 +1,119 @@
+import argparse
+import logging
+import pathlib
+import re
+import yaml
+
+
+logger = logging.getLogger(__name__)
+
+
+PIP_REQUIREMENT_FILES = {
+    'requirements.txt': ('host', 'build', 'run'),
+    'dev-requirements.txt': ('test', ),
+    # 'doc-requirements.txt': 'build',  # ?
```

Comment:
Should we continue splitting dev-requirements and doc-requirements? I could see it both ways

Reply:
Downsides for combining:
* Keeping dev+doc together increases overhead for testing when not building docs (sphinx et al are quite large)

Upsides for combining:
* Simplicity
* Clear mapping from standard conda-recipe requirements

Maybe the conda-recipe format is flexible enough that we can add a `docs:` requirement  section?  cc @hhslepicka who might know more.

Reply:
The hope being that adding a `doc:` section will be ignored for normal conda builds, but can be used to help us track and update the `doc-requirements.txt` file? Seems nice

Reply:
As an aside: a bunch of our repos have `docs-requirements.txt` instead of `doc-requirements.txt`

Reply:
Yeah, exactly on the `doc` section

Reply:
I am trying to find the conda linter. I think that `conda-forge` will flip out with that...
Building locally probably it won't be a problem. I will give it a try locally.

Reply:
Ok.. so here is what I could get tested for now:
I applied the following patch to my pydm `conda-recipe/meta.yaml`.
```
+    doc:
+      - sphinx
+    docs:
+      - sphinx_rtd_theme
+      - sphinxcontrib-napoleon
```
The `conda build` did not complain about it at all... but the `conda-forge` linter is not happy with that... See https://github.com/conda-forge/pydm-feedstock/pull/4

Reply:
Good about `conda build` - I'd guess that you're separately maintaining `meta.yaml` in the repo + conda-forge like our PCDS repos? This could still be viable, if so...

Reply:
@klauer I agree 100% here. They are different anyway... conda-forge will always have a peculiar linter and build system. Let's keep them separated. I like the way this CLI works and the output it gives.